src/ripple/shamap/impl/SHAMap.cpp

```diff
@@ -468,14 +470,37 @@ SHAMap::firstBelow(
             }
             inner = std::static_pointer_cast<SHAMapInnerNode>(node);
             stack.push({inner, stack.top().second.getChildNodeID(branch)});
-            i = 0;  // scan all 16 branches of this new node
+            i = init;  // descend and reset loop
         }
         else
-            ++i;  // scan next branch
+            incr(i);  // scan next branch
     }
     return nullptr;
 }
-
+SHAMapLeafNode*
+SHAMap::lastBelow(
+    std::shared_ptr<SHAMapTreeNode> node,
+    SharedPtrNodeStack& stack,
+    int branch) const
+{
+    return below(
+        node,
+        stack,
+        branch,
+        {branchFactor - 1, [](int i) { return i >= 0; }, [](int& i) { --i; }});
```

Comment:
This one is bikeshedding, but IMO it would be easier to read if we did something like:
```cpp
auto init = branchFactor - 1;
auto cmp = [](int i) { return i >= 0; };
auto inc = [](int& i) { --i; };

return below(node, stack, branch, {init, cmp, inc});
```
versus just passing in a bunch of lambdas in as a tuple.

Same for firstBelow, totally up to you.